packages/gatsby/src/utils/map-pages-to-static-query-hashes.ts

```diff
@@ -0,0 +1,130 @@
+import { uniqBy } from "lodash"
+import { IGatsbyState } from "../redux/types"
+
+const mapComponentsToStaticQueryHashes = (
+  staticQueryComponents: IGatsbyState["staticQueryComponents"]
+): Map<string, string> =>
+  Array.from(staticQueryComponents).reduce(
+    (map, [, { componentPath, hash }]) => {
+      map.set(componentPath, hash)
+      return map
+    },
+    new Map()
+  )
+
+export function mapTemplatesToStaticQueryHashes(
+  reduxState: IGatsbyState,
+  compilation
+): Map<string, Array<number>> {
+  const { components, staticQueryComponents } = reduxState
+  const { modules } = compilation
+
+  const terminalNodes = [
+    `.cache/api-runner-browser-plugins.js`,
+    `.cache/async-requires.js`,
+  ]
+
+  const globalStaticQueries = new Set<string>()
+
+  const getDeps = (mod): any => {
```

Comment:
Can we get some types in here?

Reply:
Webpack doesn't seem to export a Module type that actually includes everything (like .identifier) so after many attempts of looking through their code base, I kinda gave up...

Reply:
We've all found the webpack types to be a bit lacking. What I would do here though is to define a module type, based on `Webpack.compilation.Module`, that adds the missing types, and use that.

Reply:
Done